src/Microsoft.AspNetCore.SignalR/DefaultHubLifetimeManager.cs

```diff
@@ -81,13 +85,9 @@ private Task InvokeAllWhere(string methodName, object[] args, Func<ConnectionCon
             return Task.WhenAll(tasks);
         }
 
-        public override Task InvokeConnectionAsync(string connectionId, string methodName, object[] args)
+        public override IHubClientProxy GetConnectionProxy(string connectionId)
         {
-            var connection = _connections[connectionId];
-
-            var message = new InvocationMessage(GetInvocationId(), nonBlocking: true, target: methodName, arguments: args);
-
-            return WriteAsync(connection, message);
+            return _connections[connectionId].Metadata.Get<IHubClientProxy>(typeof(HubConnection));
```

Comment:
That doesn't seem so crazy to me. It's what HubConnection represents: A handle to a hub on the other peer.